### DISCUSSION TITLE
Upgrade from 2.4.1 to 2.4.7 Ubuntu

### DISCUSSION THREAD

**Discussion by terrelliott** [OpenPrinting/cups#948] [Q&A] [answered]
I feel like an idiot. I can't get my Ubuntu 22.04.4 LTS to upgrade to CUPS 2.4.7. I'm stuck at 2.4.1.
Please help...

**Comment by zdohnal** [comment]
Hi,

it would be great if you provided some errors/logs which appear when you try upgrading.

Thank you in advance!

**Comment by terrelliott** [comment]
It's not even that I am getting errors. I can't figure out the process. I've got the file in my downloads. cups-2.4.7-source.tar.gz
If I need to do a fresh install, fine, but I come from the Windows world and can't figure out the syntax to put in the commands to update. Using the Applications store in Ubuntu it says it's installing the Cups 2.4.7, but it installed 2.4.1 instead. So I am trying to install manually but having no luck

**Comment by terrelliott** [comment]
It's crashing out at attached error
![image](https://github.com/OpenPrinting/cups/assets/167825979/0ed0cd51-1a38-42b5-bfe1-ee4373e765c1)

**Reply by zdohnal** [reply]
If you come from Windows world and you are not used to configuring and compiling software in Linux, I recommend using CUPS from Ubuntu repositories - you can go to Software manager (or Synaptic, I'm not really sure - I use mostly command line for installing software and I have a different distro, so I'm not sure about the right name), find CUPS there and install it. If Ubuntu OS installs 2.4.1 instead of 2.4.7 that's something you should report to Ubuntu distro.

Otherwise, software needs its requirements to be installed before you configure and compile it. The error says you specified TLS support, but there is no development files installed for a TLS library.

**Comment by terrelliott** [comment]
I'm trying to learn, so just using the Ubuntu repositories would be ideal, but the version they have there is several major revisions behind. I'd like to obtain the skillset to work around these issues. I got as far as the error above, so I feel like that's progress, just need help resolving the issue.
That being said, the configure file is what specified TLS support. Any idea how to turn that off, or how to install TLS development files? I tried finding the TLS=Yes in the configure text file, but no luck

**Reply by michaelrsweet** [reply]
"more INSTALL.md" will show you commands to install the prerequisites...

**Reply by terrelliott** [reply]
Use that after running the "file".tar.gz command, but before ./configure?

**Reply by michaelrsweet** [reply, marked as answer]
You can also look at the online version [here](https://github.com/OpenPrinting/cups/blob/master/INSTALL.md). The commands are:

```
sudo apt-get install autoconf build-essential libavahi-client-dev \
     libgnutls28-dev libkrb5-dev libnss-mdns libpam-dev \
     libsystemd-dev libusb-1.0-0-dev zlib1g-dev
```

**Reply by terrelliott** [reply]
Thank you, that is most helpful!

**Reply by terrelliott** [reply]
Looks like 2.4.1 is still the official distro for Ubuntu. I guess they are still having issues. I can install it, but it just won't run. But thank you so much for your help!
